R/games.R

```diff
@@ -971,19 +971,7 @@ degseq <- function(..., deterministic = FALSE) {
 #' g <- sample_growing(500, citation = FALSE)
 #' g2 <- sample_growing(500, citation = TRUE)
 #'
-sample_growing <- function(n, m = 1, directed = TRUE, citation = FALSE) {
-  on.exit(.Call(R_igraph_finalizer))
-  res <- .Call(
-    R_igraph_growing_random_game, as.numeric(n), as.numeric(m),
-    as.logical(directed), as.logical(citation)
-  )
-  if (igraph_opt("add.params")) {
-    res$name <- "Growing random graph"
-    res$m <- m
-    res$citation <- citation
-  }
-  res
-}
+sample_growing <- growing_random_game_impl
```

Comment:
Moving forward, I'd rather have these functions _wrap_ than alias the impl functions. This will add flexibility to change the C layer. Also, ellipsis (but I forget if we need the `rlang::` prefix here). @maelle?

```suggestion
sample_growing <- function(n, m = 1, ..., directed = TRUE, citation = FALSE) {
  check_dots_empty()
  growing_random_game_impl(n, m, directed, citation)
}
```

Reply:
Well, the original didn't have the ellipsis, and adding it is a breaking change, right?

That said, I agree with adding the ellipsis, and we can easily auto-generate it, since I had to add the parameter list manually on the R side anyway to correct the default for `directed`.

I'll go ahead  with the ellipsis if I don't hear further concerns from you.

Reply:
I pushed a commit that includes the `...` and `check_dots_empty()` in the auto-generated `growing_random_game_impl()`. Let me know if you still want the wrapper.

If we go with the wrapper, then default parameter values are specified purely in R instead of through auto-generation. (This is not good or bad, just pointing it out.)

Reply:
Keep in mind that if we do wrappers, we will need to manually call `match.arg` in the wrapper (at least that's how I understand it). This function doesn't require `match.arg` so it's not affected.